### PR TITLE
Policy validation must return 'ok' or an error

### PR DESCRIPTION
**Pull request by dcorbacho**
Any other value will break the validation cycle and leave the rest of the arguments without validate

Reported in the mailing list: https://groups.google.com/forum/#!topic/rabbitmq-users/7mwL2AWwUyI

## Types of Changes
- [x] Bug fix (non-breaking change which fixes issue #NNNN)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)

## Checklist
- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] All tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in related repositories

## Further Comments

If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc.

### CODE REVIEW

File: src/rabbit_queue_location_validator.erl [hunk 1]
```diff
@@ -33,7 +33,10 @@
 
 validate_policy(KeyList) ->
     case proplists:lookup(<<"queue-master-locator">> , KeyList) of
-        {_, Strategy} -> validate_strategy(Strategy);
+        {_, Strategy} -> case validate_strategy(Strategy) of
```
Comment (lukebakken) [inline src/rabbit_queue_location_validator.erl]: I must be missing something, but I don't see how `validate_strategy/1` can return anything other than `{ok, _}` or `{error, "~p invalid queue-master-locator value", [Strategy]}`

Comment (hairyhum) [reply src/rabbit_queue_location_validator.erl]: `validate_policy/1` should return either `ok` or `{error, term()}`, but not `{ok, term()}`. https://github.com/rabbitmq/rabbitmq-server/blob/master/src/rabbit_policy.erl#L528

Comment (lukebakken) [reply src/rabbit_queue_location_validator.erl]: I see now, thanks.

**Comment by dcorbacho** [comment]
@hairyhum Test just added

**Comment by michaelklishin** [comment]
Backported to `v3.7.x`.

**Pull request merged**